git-gud

```diff
@@ -6,7 +6,7 @@ validateName() {
   [[ "$1" =~ $re ]]
 }
 validateEmail() {
-  local re="^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[a-z]{2,}$"
+  local re="^[a-zA-Z0-9\.]+@[a-zA-Z0-9]+\.[a-z]{2,}$"
```

Comment:
Regexes are so complex... Seems like dots are allowed in the local part of an email if there are no two consecutive dots and also no leading or trailing dots 🤔 Can you check if this work for you?

`^(?!.*\.\.)(?!.*\.$)\w*@[a-zA-Z0-9]+\.[a-z]{2,}$`

Reply:
That one did not work, `^(?!.*\.\.)^[a-zA-Z0-9][a-zA-Z0-9\.]*@[a-zA-Z0-9]+\.[a-z]{2,}$` seems to work with the cornercases you've described.

![image](https://github.com/user-attachments/assets/05fe4112-aeee-4613-bc14-3b1e35216613)

Reply:
Fixed.

Reply:
Seems like a trailing dot is still allowed but should not be allowed (`something.@something.com` must be rejected). Maybe `^(?!.*\.\.)^[a-zA-Z0-9\.]*[a-zA-Z0-9]@[a-zA-Z0-9]+\.[a-z]{2,}$` instead?

Reply:
Didn't notice that the regex in your PR already addressees this, and it seems to be ok so let's accept that 👌